assembly/src/parser/error.rs

```diff
@@ -140,9 +140,20 @@ pub enum ParsingError {
         span: SourceSpan,
         expected: Vec<String>,
     },
+    #[error("{error}")]
+    #[diagnostic(help(
+        "bare identifiers must be lowercase alphanumeric with '_', quoted identifiers can include any graphical character"
+    ))]
+    InvalidIdentifier {
+        #[source]
+        #[diagnostic(source)]
+        error: crate::ast::IdentError,
+        #[label]
+        span: SourceSpan,
+    },
     #[error("invalid character in identifier")]
     #[diagnostic(help(
-        "bare identifiers must be lowercase alphanumeric with '_', quoted identifiers can include uppercase, as well as '.' and '$'"
+        "bare identifiers must be lowercase alphanumeric with '_', quoted identifiers can include any graphical character"
     ))]
     InvalidIdentCharacter {
```

Comment:
This variatn is unused (in the assembler at least) and is redundant with `IdentError::InvalidChars` - should we remove it?